TEST: Static method in PanacheEntity with parameters type 'array of primitive' generate error during deployment phase

### Describe the bug

Error in deployment phase when launching test caused by method enhancement of Panache's static methods which signature contains `array of primitive` parameter(s).
Error is present only during `test` s because the problem shows up only adding `io.quarkus:quarkus-panache-mock` dependency

### Expected behavior

No errors

### Actual behavior

```
Caused by: java.lang.IllegalArgumentException: Not a valid array name
	at org.jboss.jandex.Type.create(Type.java:144)
	at io.quarkus.deployment.util.AsmUtil.getParameterTypes(AsmUtil.java:707)
	at io.quarkus.panache.common.deployment.visitors.PanacheEntityClassOperationGenerationVisitor.visitMethod(PanacheEntityClassOperationGenerationVisitor.java:94)
	at org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1353)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:744)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:424)
	at io.quarkus.deployment.steps.ClassTransformingBuildStep.transformClass(ClassTransformingBuildStep.java:321)
	at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:205)
	at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:192)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### How to Reproduce?

[Link to reproducer](https://github.com/luca-bassoricci/panache-method-visitor)
Run test

### Output of `uname -a` or `ver`

Microsoft Windows [Versione 10.0.19042.1469]

### Output of `java -version`

openjdk version "11.0.6" 2020-01-14 LTS

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

### Additional information

Version `2.7.4.Final` doesn't solve the problem

My problems are related to vertx-http module: tests fail with this stacktrace:
quarkus-vertx-http-deployment --- TESTS
```
Caused by: java.lang.RuntimeException: Unable to start HTTP server
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:625)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:282)
        at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket1866188241.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket1866188241.deploy(Unknown Source)
        ... 50 more
Caused by: java.util.concurrent.ExecutionException: io.quarkus.runtime.QuarkusBindException
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:591)
        ... 53 more
Caused by: io.quarkus.runtime.QuarkusBindException
```
and I'm pretty sure no other services are running on port 8080 :/
any advice?

--- a/core/deployment/src/main/java/io/quarkus/deployment/util/AsmUtil.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/util/AsmUtil.java
@@ -61,11 +61,16 @@ public class AsmUtil {
             getType(Long.class),
             getType(Double.class));
     public static final Map<org.objectweb.asm.Type, org.objectweb.asm.Type> WRAPPER_TO_PRIMITIVE = new HashMap<>();
+    public static final Map<Character, String> PRIMITIVE_DESCRIPTOR_TO_PRIMITIVE_CLASS_LITERAL;
 
     static {
         for (int i = 0; i < AsmUtil.PRIMITIVES.size(); i++) {
             AsmUtil.WRAPPER_TO_PRIMITIVE.put(AsmUtil.WRAPPERS.get(i), AsmUtil.PRIMITIVES.get(i));
         }
+        PRIMITIVE_DESCRIPTOR_TO_PRIMITIVE_CLASS_LITERAL = Map.of(
+                'Z', "boolean", 'B', "byte", 'C', "char",
+                'D', "double", 'F', "float", 'I', "int",
+                'J', "long", 'S', "short");
     }
 
     public static org.objectweb.asm.Type autobox(org.objectweb.asm.Type primitive) {
@@ -698,50 +703,22 @@ public class AsmUtil {
             char c = chars[i];
             switch (c) {
                 case 'Z':
-                    args.add(Type.create(DotName.createSimple("boolean"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'B':
-                    args.add(Type.create(DotName.createSimple("byte"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'C':
-                    args.add(Type.create(DotName.createSimple("char"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'D':
-                    args.add(Type.create(DotName.createSimple("double"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'F':
-                    args.add(Type.create(DotName.createSimple("float"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'I':
-                    args.add(Type.create(DotName.createSimple("int"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'J':
-                    args.add(Type.create(DotName.createSimple("long"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'S':
-                    args.add(Type.create(DotName.createSimple("short"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
+                    final Type type;
+                    if (dimensions == 0) {
+                        type = Type.create(DotName.createSimple(PRIMITIVE_DESCRIPTOR_TO_PRIMITIVE_CLASS_LITERAL.get(c)),
+                                Kind.PRIMITIVE);
+                    } else {
+                        DotName dotName = DotName.createSimple("[".repeat(dimensions) + c);
+                        type = Type.create(dotName, Kind.ARRAY);
+                    }
+                    args.add(type);
                     dimensions = 0;
                     start = i + 1;
                     break;

--- a/core/deployment/src/test/java/io/quarkus/deployment/util/AsmUtilTest.java
+++ b/core/deployment/src/test/java/io/quarkus/deployment/util/AsmUtilTest.java
@@ -1,0 +1,41 @@
+package io.quarkus.deployment.util;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.jboss.jandex.Type;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class AsmUtilTest {
+
+    @ParameterizedTest
+    @MethodSource
+    void testGetParameterTypes(String methodDescriptor, String... expected) {
+        assertArrayEquals(expected,
+                Stream.of(AsmUtil.getParameterTypes(methodDescriptor)).map(Type::toString).toArray(String[]::new));
+    }
+
+    private static final Stream<Arguments> testGetParameterTypes() {
+        List<Arguments> arguments = new ArrayList<>();
+        final var array1 = new StringBuilder();
+        final var array2 = new StringBuilder();
+        for (int i = 0; i < 5; i++) {
+            for (Character c : AsmUtil.PRIMITIVE_DESCRIPTOR_TO_PRIMITIVE_CLASS_LITERAL.keySet()) {
+                arguments.add(Arguments.of("(" + array2 + c + ")V",
+                        toArray(AsmUtil.PRIMITIVE_DESCRIPTOR_TO_PRIMITIVE_CLASS_LITERAL.get(c) + array1)));
+            }
+            array1.append("[]");
+            array2.append('[');
+        }
+        return arguments.stream();
+    }
+
+    private static String[] toArray(String... values) {
+        return values;
+    }
+}

